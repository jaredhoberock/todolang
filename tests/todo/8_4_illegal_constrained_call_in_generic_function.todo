fun foo<T: Add>(arg: T) -> Number {
  0
}

fun bar<T>(arg: T) -> Number {
  // XXX this call should fail because a constraint couldn't be solved
  foo(arg)
}
