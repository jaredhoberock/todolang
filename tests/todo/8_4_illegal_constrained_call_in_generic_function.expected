[0m[1m[38;5;9merror[0m[1m: The trait bound 'T3: Add' is not satisfied'[0m
  [0m[34mâ”Œâ”€[0m tests/todo/8_4_illegal_constrained_call_in_generic_function.todo:7:7
  [0m[34mâ”‚[0m
[0m[34m7[0m [0m[34mâ”‚[0m   foo([0m[31marg[0m)
  [0m[34mâ”‚[0m   [0m[34m---[0m [0m[31m^^^[0m
  [0m[34mâ”‚[0m   [0m[34mâ”‚[0m   [0m[34mâ”‚[0m
  [0m[34mâ”‚[0m   [0m[34mâ”‚[0m   [0m[34mthe trait 'Add' is not implemented for 'T3'[0m
  [0m[34mâ”‚[0m   [0m[34mrequired by a bound introduced by this call[0m


